ui/test/nuxeo-lock-toggle-button.test.js

```diff
@@ -14,82 +14,318 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 */
-import { fixture, html, login, waitChanged } from '@nuxeo/nuxeo-elements/test/test-helpers.js';
+import { fixture, flush, html, login, waitForEvent } from '@nuxeo/nuxeo-elements/test/test-helpers.js';
 import { tap } from '@polymer/iron-test-helpers/mock-interactions.js';
+import { dom } from '@polymer/polymer/lib/legacy/polymer.dom';
 import '../actions/nuxeo-lock-toggle-button.js';
 /* eslint-disable no-unused-expressions */
 
 suite('<nuxeo-lock-toggle-button>', () => {
+  let button;
   let server;
+
+  const lockedDocument = {
+    'entity-type': 'document',
+    uid: '1',
+    contextParameters: {
+      permissions: ['Write'],
+    },
+    facets: [],
+    lockCreated: '2016-02-19T12:47:40.501+01:00',
+    lockOwner: 'Administrator',
+    type: 'File',
+  };
+
   setup(async () => {
     server = await login();
+    button = await fixture(
+      html`
+        <nuxeo-lock-toggle-button></nuxeo-lock-toggle-button>
+      `,
+    );
   });
 
-  suite('when a document is locked', () => {
-    let element;
-    setup(async () => {
-      const doc = {
-        'entity-type': 'document',
-        uid: '1',
-        lockCreated: '2016-02-19T12:47:40.501+01:00',
-        lockedBy: 'Administrator',
-        facets: [],
-        type: 'File',
+  suite('Button Visibility', () => {
+    test('Should not be visible when document is null', async () => {
+      button.document = null;
+      await flush();
+      expect(dom(button.root).querySelector('.action')).to.be.null;
```

Comment:
I've got mixed feelings about this check tbh:
1. If we switch the action to just hide stuff then this test will fail
2. since the dom-if is not set to restamp it will fail at some point (dom-if hides things after the first stamp for performance)
3. might as well just unit test the `_isAvailable` method and pass the doc (no need for fixtures)

Reply:
Very valid point! Indeed, in this specific case testing the `_isAvailable` would be enough.
Anyway, we still need to think how to test visibilities in general because we have elements that need to inspect the DOM.